server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -233,6 +237,24 @@ public void addNamedQuery(String name, Query query) {
         return unmodifiableMap(new HashMap<>(namedQueries));
     }
 
+    public ParsedDocument parseDocument(String type, SourceToParse source) throws MapperParsingException {
```

Comment:
@romseygeek I had to add a type argument here on 7.x ,does that look ok?

Reply:
++